pkg/proxy/ipvs/proxier.go

```diff
@@ -856,20 +856,30 @@ func CleanupLeftovers(ipvs utilipvs.Interface, ipt utiliptables.Interface, ipset
 			encounteredError = true
 		}
 	}
-	// Delete dummy interface created by ipvs Proxier.
+	// Delete dummy interface created by ipvs Proxier. Should delete dummy interface once it exist.
 	nl := NewNetLinkHandle()
-	err := nl.DeleteDummyDevice(DefaultDummyDevice)
-	if err != nil {
-		glog.Errorf("Error deleting dummy device %s created by IPVS proxier: %v", DefaultDummyDevice, err)
-		encounteredError = true
+	if found := nl.IsDummyDeviceExist(DefaultDummyDevice); found {
+		err := nl.DeleteDummyDevice(DefaultDummyDevice)
+		if err != nil {
+			glog.Errorf("Error deleting dummy device %s created by IPVS proxier: %v", DefaultDummyDevice, err)
+			encounteredError = true
+		}
 	}
+
 	// Clear iptables created by ipvs Proxier.
 	encounteredError = cleanupIptablesLeftovers(ipt) || encounteredError
 	// Destroy ip sets created by ipvs Proxier.  We should call it after cleaning up
 	// iptables since we can NOT delete ip set which is still referenced by iptables.
 	ipSetsToDestroy := []string{KubeLoopBackIPSet, KubeClusterIPSet, KubeLoadBalancerSet, KubeNodePortSetTCP, KubeNodePortSetUDP,
 		KubeExternalIPSet, KubeLoadBalancerSourceIPSet, KubeLoadBalancerSourceCIDRSet, KubeLoadBalancerMasqSet}
-	for _, set := range ipSetsToDestroy {
+	entryIPSetIn := []string{string(utilipset.HashIPPortIP), string(utilipset.HashIPPort), string(utilipset.HashIPPort), string(utilipset.HashIPPort), string(utilipset.HashIPPort),
+		string(utilipset.HashIPPortIP), string(utilipset.HashIPPortNet), string(utilipset.BitmapPort), string(utilipset.BitmapPort)}
+	for i, set := range ipSetsToDestroy {
+		found, err := ipset.TestEntry(entryIPSetIn[i], set)
```

Comment:
Please double check this line, is it really a right fix?

The 1st parameter of `ipset.TestEntry()` is entry name instead of entry type. And, according the error log, we should test the exist of ip set instead of entry?

Reply:
I will check at once. It seems that no command for just test one  ipset exist, i have run `ipset --help`. Have you any good suggestion?

Reply:
Can we parse the error message and just ignore the not found error?